csv-to-genesis/src/main/kotlin/jp/co/soramitsu/Converter.kt

```diff
@@ -0,0 +1,153 @@
+        val client = Iroha2Client(peerUrl, true)
+        csv.bufferedReader().use { reader ->
+            val parser = CSVParser(
+                reader, CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withSkipHeaderRecord()
+            )
+            parser.forEach {
+                client.send(
+                    *it.mapToAssetIsi(admin).toTypedArray(),
+                    account = admin,
+                    keyPair = keyPair
+                )
+                println("#${it.get(0)} saved")
+            }
+        }
+    }
+
+    fun toGenesis(csv: File, genesis: File, admin: AccountId): Genesis {
+        val node = JSON_SERDE.readTree(genesis)
+        val block = JSON_SERDE.convertValue(node, RawGenesisBlock::class.java)
+        val txs = block.transactions.cast<MutableList<GenesisTransaction>>()
+
+        csv.bufferedReader().use { reader ->
+            val parser = CSVParser(
+                reader, CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withSkipHeaderRecord()
+            )
+            val isi = mutableListOf<Instruction>()
+            parser.forEach { isi.addAll(it.mapToAssetIsi(admin)) }
+            txs.add(GenesisTransaction(isi))
+        }
+
+        return Genesis(RawGenesisBlock(txs))
+    }
+
+    private fun CSVRecord.mapToAssetIsi(accountId: AccountId): ArrayList<Instruction> {
+        val isi = mutableListOf<Instruction>()
+
+        val id = this.get(ID.second)
+        val definitionId = AssetDefinitionId(id.asName(), accountId.domainId)
+        val assetId = AssetId(definitionId, accountId)
+
+        Instructions.registerAsset(definitionId, AssetValueType.Store())
+            .also { isi.add(it) }
+        Instructions.setKeyValue(definitionId, ID.first, id.asValue())
+            .also { isi.add(it) }
+
+        this.toSkvIsi<Long>(assetId, FRAUD_TYPE).also { isi.add(it) }
+        this.toSkvIsi<String>(assetId, ORIGINATION).also { isi.add(it) }
+        this.toSkvIsi<String>(assetId, DESTINATION).also { isi.add(it) }
+        this.toSkvIsi<Long>(assetId, STATUS).also { isi.add(it) }
+        this.toSkvIsi<Date>(assetId, TIMESTAMP).also { isi.add(it) }
+
+        dateFormatter
+            .parse(this.get(EXPIRY_DATE.second))
+            .toEpochSeconds(Duration.ofDays(120))
+            .asValue()
+            .let { date -> Instructions.setKeyValue(assetId, EXPIRY_DATE.first, date) }
+            .also { isi.add(it) }
+
+        Instructions.grantSetKeyValueAsset(assetId, accountId).also { isi.add(it) }
+
+        return ArrayList(isi)
+    }
+
+    private suspend fun Iroha2Client.send(
+        vararg isi: Instruction,
+        account: AccountId,
+        keyPair: KeyPair
+    ) = TransactionBuilder {
+        account(account)
+        this.instructions.value.addAll(isi)
+    }.let { builder ->
+        this.fireAndForget {
```

Comment:
I guess it'd better to send a single ultimate transaction with blocking manner to wait a response to verify everything is ok